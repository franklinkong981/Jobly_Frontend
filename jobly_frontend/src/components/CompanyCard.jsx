import React from "react";

function CompanyCard() {
  return (

  );
}

export default CompanyCard;