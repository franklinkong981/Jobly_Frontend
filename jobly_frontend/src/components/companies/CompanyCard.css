.CompanyCard {
  border: 2px solid red;
  background-color: lightgreen;
}