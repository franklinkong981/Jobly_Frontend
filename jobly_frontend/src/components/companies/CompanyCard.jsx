import React from "react";
import { Link } from "react-router-dom";

import "./CompanyCard.css";

/**
 * Shows information about a company.
 * Rendered by CompanyList to show a card for each company.
 * Contains the company's name, description, and a link to the company details page.
 */
function CompanyCard({id, name, description}) {

  return (
    <section className="CompanyCard card">
      <div className="Companycard-content card-body">
        <h3 className="CompanyCard-title card-title">
          {name}
        </h3>
        <p className="CompanyCard-description">{description}</p>
        <Link className="CompanyCard-link btn btn-info font-weight-bold" to={`/companies/${id}`}>
          See Job Openings
        </Link>
      </div>
    </section>
  );
}

export default CompanyCard;
