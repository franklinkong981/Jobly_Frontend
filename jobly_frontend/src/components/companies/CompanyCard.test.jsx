import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import CompanyCard from "./CompanyCard.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <CompanyCard id="rithm" name="Rithm School" description="Become an exceptional developer in 16 weeks." />
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <CompanyCard id="rithm" name="Rithm School" description="Become an exceptional developer in 16 weeks." />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});