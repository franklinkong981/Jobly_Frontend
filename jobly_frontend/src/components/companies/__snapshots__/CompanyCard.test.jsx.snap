// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="CompanyCard card"
  >
    <div
      class="Companycard-content card-body"
    >
      <h3
        class="CompanyCard-title card-title"
      >
        Rithm School
      </h3>
      <p
        class="CompanyCard-description"
      >
        Become an exceptional developer in 16 weeks.
      </p>
      <a
        class="CompanyCard-link btn btn-info font-weight-bold"
        data-discover="true"
        href="/companies/rithm"
      >
        See Job Openings
      </a>
    </div>
  </section>
</DocumentFragment>
`;
